cli/commands/preview/tests/create.test.ts

```diff
@@ -0,0 +1,80 @@
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import nock from 'nock';
+import { PreviewCreateCommand } from 'cli/commands/preview/create';
+
+jest.mock( 'fs' );
+
+describe( 'PreviewCreateCommand', () => {
+	let command: PreviewCreateCommand;
+	const mockFolder = '/test/folder';
+	const mockArchivePath = path.join( os.tmpdir(), `${ mockFolder }.zip` );
+
+	beforeEach( () => {
+		jest.clearAllMocks();
+		nock.cleanAll();
+		command = new PreviewCreateCommand( mockFolder, 'json' );
+	} );
+
+	describe( 'run', () => {
+		it( 'should complete the preview creation process successfully', async () => {
+			const mockWriteStream = {
+				on: jest.fn( ( event, callback ) => {
+					if ( event === 'close' ) {
+						callback();
+					}
+				} ),
+			};
+			( fs.createWriteStream as jest.Mock ).mockReturnValue( mockWriteStream );
+
+			nock( 'https://public-api.wordpress.com' )
+				.post( '/rest/v1.1/jurassic-ninja/create-new-site-from-zip' )
+				.reply( 200, { success: true } );
+
+			( fs.unlinkSync as jest.Mock ).mockImplementation( () => {} );
+
+			const result = await command.run();
+
+			expect( result ).toBe( true );
+			expect( fs.createWriteStream ).toHaveBeenCalledWith( mockArchivePath );
+			expect( fs.unlinkSync ).toHaveBeenCalledWith( mockArchivePath );
+		} );
+	} );
+
+	describe( 'archiveFolder', () => {
```

Comment:
I feel ideally we should be testing private functions as these are implementation details that can change, the important big being the "run" function.